Source/CommandsWindow.cpp

```diff
@@ -173,6 +181,32 @@ CommandsWindow::_BrowseCommand()
 }
 
 
+void
+CommandsWindow::_EditCommand()
+{
+	if (_CommandIsScript()) {
+		const char* argv[] = { fCommandControl->Text(), NULL };
+		be_roster->Launch("text/x-source-code", 1, argv);
+	}
+}
+
+
+bool
+CommandsWindow::_CommandIsScript()
+{
+	BPath path = fCommandControl->Text();
+	if (path.InitCheck() != B_OK)
+		return false;
+
+	char mimeType[B_MIME_TYPE_LENGTH];
+	BNode commandNode(path.Path());
+	BNodeInfo(&commandNode).GetType(mimeType);
+	if (strncmp("text/", mimeType, 5) == 0)
+		return true;
+
+	return false;
+}
+
```

Comment:
This is going to cause a problem if `fCommandControl` has shell escapes/quotes in the command or command line arguments.  For example, if the command was `/boot/home/myscript.sh --myarg` then it would need to have the `--myarg` removed.  I have some **really ugly and crude** code at https://github.com/augiedoggie/TrackRunner/blob/026812517d05a2a8eeb6f26adc46bf210891caab/Source/RunnerAddOn.cpp#L191 that could be moved to its own function and reused but I'd like to find a more reliable way to extract just the file path from the whole command string.

Reply:
Haven't thought about that...
It'd be easiest and probably most robust to split that in two BTextControls:

```
Command/Script:    /boot/home/myscript.sh
Arguments:         --myarg 5038 --myarg2 etc.
```

Or do you deem that too crude... :)

Reply:
Certainly I've thought about splitting it up, and I may end up doing that eventually, but I was really avoiding doing that because I think it makes the user interface more cumbersome to use.  Much easier to see or type out a whole command in one input box.  I'll have to think about it to see if there are any other consequences of changing it.

Reply:
One problem is going to be migrating existing users to the new system.  It would be difficult to automatically convert the old commands to the split version.

Reply:
True. Good luck with your command/argument parse&split then... :)